client/components/platform-checkout/save-user/phone-number-input.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import React, { useEffect, useState } from 'react';
+import { TextControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import intlTelInput from 'intl-tel-input';
+
+/**
+ * Internal dependencies
+ */
+// eslint-disable-next-line import/no-unresolved
+import utils from 'iti/utils';
+
+const PhoneNumberInput = ( { handlePhoneNumberChange } ) => {
+	const [ inputValue, setInputValue ] = useState( '' );
+	const [ inputInstance, setInputInstance ] = useState( null );
+
+	const handlePhoneNumberInputChange = ( value = inputValue ) => {
+		setInputValue( value );
+		if ( inputInstance ) {
+			handlePhoneNumberChange( inputInstance.getNumber() );
+		}
+	};
+
+	useEffect( () => {
+		let iti = null;
+		const input = document.querySelector( '#phone-number' );
+
+		const handleCountryChange = () => {
+			handlePhoneNumberChange( iti.getNumber() );
+		};
+
+		if ( input ) {
+			iti = intlTelInput( input, {
+				initialCountry: 'US',
+				customPlaceholder: function ( selectedCountryPlaceholder ) {
+					return selectedCountryPlaceholder;
+				},
+				separateDialCode: true,
+				utilsScript: utils,
+			} );
+			setInputInstance( iti );
+
+			input.addEventListener( 'countrychange', handleCountryChange );
+		}
+
+		return () => {
+			if ( iti ) {
+				iti.destroy();
+				input.removeEventListener(
+					'countrychange',
+					handleCountryChange
+				);
+			}
+		};
+	}, [ handlePhoneNumberChange ] );
+
+	return (
+		<TextControl
+			type="tel"
+			id="phone-number"
```

Comment:
```suggestion
			aria-label={ __( 'Mobile phone number', 'woocommerce-payments' ) }
```
I think I figured out why that one JS test (`should render the save user form when checkbox is checked`) keeps failing! The input is there, but Jest is unable to locate it.

In the test, we use `screen.queryByLabelText( 'Mobile phone number' )` to find the rendered phone number input. This finds the label containing the described text and then locates its associated input. [Here](https://testing-library.com/docs/queries/bylabeltext) are the patterns that are used to connect label to input. However, `intl-tel-input` appears to do things a little differently, when we feed it our phone input. Here's what it looks like in the HTML source.

![Phone number HTML](https://cdn-std.droplr.net/files/acc_1104206/62ZwsB)
_`intl-tel-input` rendered HTML_

Note that our label has no `id` property, its `for` property does not match the `id` nor `name` on the input, and the input is sitting outside the label: none of the patterns match what Jest is scanning for, so `screen.queryByLabelText( 'Mobile phone number' )` returns `null`!

I added the property `aria-label` to the input (as you can see in the screenshot above) and now Jest can find the input and the test passes at last! Try it out and let me know if that works for you too.

This does raise another issue that perhaps you have some more thoughts on: the mobile phone number label does not really work as an HTML label at the moment. Its `for` property does not match the `name` nor `id` on the input, so when we click on the label, the phone number input isn't automatically focussed on, as one might expect. 

If we look at the [source code for this `TextControl` component](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/text-control/index.js#L44), it seems that the `id` property which becomes the label's `id` is generated dynamically, whereas the `id` property for the input can be overwritten by our own properties on the component. The only way to keep the `id` on the label and the input the same is to remove the `id` field from the list of properties when we render the `TextControl` element and allow both ids to be generated dynamically.

But if we do that, now we have no selector for the input to feed to the initial `intlTelInput` function to create our magic phone input! I used the below as a substitute selector--and it works--but it is a little ugly.

```js
const input = document.querySelector( `input[aria-label="${ __( 'Mobile phone number', 'woocommerce-payments' ) }"]` );
iti = intlTelInput( input, {
				initialCountry: 'US',
				customPlaceholder: function ( selectedCountryPlaceholder ) {
					return selectedCountryPlaceholder;
				},
				separateDialCode: true,
				utilsScript: utils,
			} );
```

Now the phone input renders correctly _and_ clicking on the label selects the input as well (and the tests pass)! Let me know your thoughts and if you have any smarter ideas for a cleaner solution here! We could add a `className` property to the `TextControl` element as well and then use something like `.className input` to select the input, but I'll leave it up to you to decipher to best solution to this puzzling predicament.

Reply:
Thanks for the suggestion @FangedParakeet :+1: 

I also figured out the first part that `int-tel-input` is changing the hierarchy of `label` and `input`. For that, I could no longer find the element with `labelText` and had to use the `test-id` instead. :eyes: 

The idea of `area-label` is a good finding and solves the issue. I also did not realize that adding an explicit `id` to the `input` field is detaching the `label` from the `input`. :woman_facepalming: 

I am updating the PR with your suggestions :tada: